.github/workflows/build.yml

```diff
@@ -4,43 +4,130 @@ on:
   push:
     branches:
       - main
+      - release-*
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
   pull_request:
-  schedule:
-    - cron: '0 0 1 * *'
+    branches:
+      - main
+      - release-*
   workflow_dispatch:
 
 concurrency:
-  group: ${{ github.workflow }}-${{ (github.ref == 'refs/heads/main' && github.run_number) || github.ref }}
+  group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
 jobs:
-  build:
+  lint:
     runs-on: ubuntu-latest
+    env:
+        RUST_LOG: info
     steps:
-      - name: Checkout Repository
-        uses: actions/checkout@v4
+      - uses: actions/checkout@v4
+        name: Checkout Repository
 
-      - name: Enable Rust Caching
-        uses: Swatinem/rust-cache@v2
+      - uses: Swatinem/rust-cache@v2
+        name: Enable Rust Caching
         with:
           cache-on-failure: true
 
       - name: Format Check
         run: cargo fmt -- --check
 
       - name: Clippy
-        run: |
-          cargo clippy --workspace -- -D warnings
+        uses: actions-rs-plus/clippy-check@v1
+        with:
+            token: ${{ github.token }}
+            args: --workspace --all-features --all-targets -- -D warnings
 
       - name: Clippy without default features
-        run: |
-          cargo clippy --no-default-features --workspace -- -D warnings
+        uses: actions-rs-plus/clippy-check@v1
+        with:
+            token: ${{ github.token }}
+            args: --workspace --no-default-features --all-targets -- -D warnings
+
+  build:
+    runs-on: ubuntu-latest
+    env:
+        RUST_LOG: info
+    steps:
+      - uses: actions/checkout@v4
+        name: Checkout Repository
+
+      - uses: Swatinem/rust-cache@v2
+        name: Enable Rust Caching
 
-      - name: Audit
-        run: cargo audit
+      - name: Build
+        run: cargo build --all-features --all-targets --release
+
+  test:
+    runs-on: ubuntu-latest
+    env:
+        RUST_LOG: info
+    steps:
+      - uses: actions/checkout@v4
+        name: Checkout Repository
+
+      - uses: Swatinem/rust-cache@v2
+        name: Enable Rust Caching
+
+      - name: Build tests
+        run: cargo test --workspace --release --all-features --no-run
 
       - name: Test
-        run: |
-          cargo test --workspace --release --no-run
-          cargo test --workspace --release --verbose -- --test-threads 2
+        run: cargo test --workspace --release --all-features --verbose -- --test-threads 2
         timeout-minutes: 60
```

Comment:
Seems excessive. Looking at the history this test usually finishes in under a minute.